src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -636,6 +645,39 @@ DbResultSet dbResultSet
             return resultProperties;
         }
 
+        /// <summary>
+        /// Reads data into jsonString.
+        /// </summary>
+        /// <param name="dbDataReader">DbDataReader.</param>
+        /// <param name="availableSize">Available buffer.</param>
+        /// <param name="jsonString">jsonString to read into.</param>
+        /// <returns>size of data read in bytes.</returns>
+        public int StreamData(DbDataReader dbDataReader, int availableSize, StringBuilder jsonString)
+        {
+            int fieldSize = (int)dbDataReader.GetChars(0, 0, null, 0, 0);
```

Comment:
named params for GetChars(...)
```suggestion
            int resultFieldSize = (int)dbDataReader.GetChars(0, 0, null, 0, 0);
```

Reply:
what happens when `dbDataReader.GetChars` returns a value outside the range of `int`?

Reply:
Fixed this to make it long. In the validate, we should handle if the result returned by getchars is greater than max int number of bytes (roughly 2GB) as our maximum limit in any case would be 158MB.